src/generateSpec.ts

```diff
@@ -302,21 +302,26 @@ function getParamSchema(
   const { explicitType, index, object, method } = param
 
   const type = Reflect.getMetadata('design:paramtypes', object, method)[index]
+  if (_.isFunction(type) && type.name === 'Array') {
+    const items = explicitType
+      ? { $ref: '#/components/schemas/' + explicitType.name }
+      : { type: 'object' }
+    return { items, type: 'array' }
+  }
+  if (explicitType) {
+    return { $ref: '#/components/schemas/' + explicitType.name }
+  }
   if (_.isFunction(type)) {
     if (_.isString(type.prototype) || _.isSymbol(type.prototype)) {
       return { type: 'string' }
     } else if (_.isNumber(type.prototype)) {
       return { type: 'number' }
     } else if (_.isBoolean(type.prototype)) {
       return { type: 'boolean' }
-    } else if (type.name === 'Array') {
-      const items = explicitType
-        ? { $ref: '#/components/schemas/' + explicitType.name }
-        : { type: 'object' }
-      return { items, type: 'array' }
     } else if (type.name !== 'Object') {
       return { $ref: '#/components/schemas/' + type.name }
```

Comment:
I think we could avoid some of these duplicate checks with something like

```typescript
  if (_.isFunction(type)) {
   if (explicitType) {
      return type.name === 'Array' ? // etc... 
    } else if
   // ...
    } else if (type.name === 'Array') {
      return { items: { type: 'object' }, type: 'array' }
    }
}
```

Reply:
Actually it should use `$ref` even if type of param is not a function. 
So if we have explicit type it should use it to define `$ref` unless the param type is an Array.